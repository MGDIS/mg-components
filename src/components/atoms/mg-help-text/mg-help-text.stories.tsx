import { h } from "@stencil/core";

export default {
  component: 'mg-help-text',
  title: 'Atoms/mg-help-text',
  parameters: {
    docs: {
      source: {
        code: `<mg-help-text identifier="identifier">
        Help text with html <strong>bold</strong>, <em>italic</em>.
</mg-help-text>`,
      },
    },
  },
};

const Template = args => <mg-help-text {...args} innerHTML={args.slot}></mg-help-text>;

export const MgHelpText = Template.bind({});
MgHelpText.args = {
  slot: 'Help text with html <strong>bold</strong>, <em>italic</em>.',
  identifier: 'identifier',
};
