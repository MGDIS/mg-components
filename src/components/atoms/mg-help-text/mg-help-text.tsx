import { Component, Host, h, Prop } from '@stencil/core';

@Component({
  tag: 'mg-help-text',
  styleUrl: 'mg-help-text.scss',
  scoped: true,
})
export class MgHelpText {

  // TODO : CHECK IF THIS COMPONENT IS REALLY USEFULL

  /**
   * Sets an `id` element.
   * Needed by the input for accessibility `arai-decribedby`.
   */
  @Prop() identifier: string;

  render() {
    return (
      <Host>
        <p id={this.identifier}><slot></slot></p>
      </Host>
    );
  }

}
