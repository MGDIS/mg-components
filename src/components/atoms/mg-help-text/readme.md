# mg-help-text



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute    | Description                                                                      | Type     | Default     |
| ------------ | ------------ | -------------------------------------------------------------------------------- | -------- | ----------- |
| `identifier` | `identifier` | Sets an `id` attribute. Needed by the input for accessibility `arai-decribedby`. | `string` | `undefined` |


## Dependencies

### Used by

 - [mg-input-checkbox](../../molecules/inputs/mg-input-checkbox)
 - [mg-input-date](../../molecules/inputs/mg-input-date)
 - [mg-input-numeric](../../molecules/inputs/mg-input-numeric)
 - [mg-input-select](../../molecules/inputs/mg-input-select)
 - [mg-input-text](../../molecules/inputs/mg-input-text)
 - [mg-input-textarea](../../molecules/inputs/mg-input-textarea)

### Graph
```mermaid
graph TD;
  mg-input-checkbox --> mg-help-text
  mg-input-date --> mg-help-text
  mg-input-numeric --> mg-help-text
  mg-input-select --> mg-help-text
  mg-input-text --> mg-help-text
  mg-input-textarea --> mg-help-text
  style mg-help-text fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
