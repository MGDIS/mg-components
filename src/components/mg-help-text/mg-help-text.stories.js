export default {
  component: 'mg-text-help',
  title: 'Atoms/mg-text-help',
  parameters: {
    docs: {
      source: {
        code: `<mg-text-help reference="reference-id">
        Help text with <a href="https://www.exemple.com">a link</a>, a <strong>bold text</strong> and <em>an italic one</em>.
</mg-text-help>`,
      },
    },
  },
};

const Template = args => <mg-text-help {...args} innerHTML={args.slot}></mg-text-help>;

export const MgCharacterLeft = Template.bind({});
MgCharacterLeft.args = {
  slot: 'Help text with <a href="https://www.exemple.com">a link</a>, a <strong>bold text</strong> and <em>an italic one</em>.',
  reference: 'reference-id',
};
