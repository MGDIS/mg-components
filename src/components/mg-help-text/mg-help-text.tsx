import { Component, Host, h, Prop } from '@stencil/core';

@Component({
  tag: 'mg-help-text',
  styleUrl: 'mg-help-text.scss',
  scoped: true,
})
export class MgHelpText {

  // TODO : CHECK IF THIS COMPONENT IS REALLY USEFULL

  /**
   * Sets an `id` element.
   * Needed by the input for accessibility `arai-decribedby`.
   */
  @Prop() reference: string;

  render() {
    return (
      <Host>
        <p id={this.reference}><slot></slot></p>
      </Host>
    );
  }

}
