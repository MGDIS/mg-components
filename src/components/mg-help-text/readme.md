# mg-help-text



<!-- Auto Generated Below -->


## Properties

| Property    | Attribute   | Description                                                                    | Type     | Default     |
| ----------- | ----------- | ------------------------------------------------------------------------------ | -------- | ----------- |
| `reference` | `reference` | Sets an `id` element. Needed by the input for accessibility `arai-decribedby`. | `string` | `undefined` |


## Dependencies

### Used by

 - [mg-input-text](../mg-input-text)

### Graph
```mermaid
graph TD;
  mg-input-text --> mg-help-text
  style mg-help-text fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
