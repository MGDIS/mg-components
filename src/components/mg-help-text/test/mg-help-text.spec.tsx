import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { MgHelpText } from '../mg-help-text';

const getPage = (args, slot) => newSpecPage({
  components: [MgHelpText],
  template: () => (<mg-help-text {...args}>{...slot}</mg-help-text>),
});

describe('mg-help-text',()=> {

  describe.each(['Simple help text.', 'Help text with <a href="https://www.exemple.com">a link</a>, a <strong>bold text</strong> and <em>an italic one</em>.'])('Should render text : %s', (helpText) => {
    test.each([undefined, 'blu'])('renders', async (reference) => {
      const { root } = await getPage({reference}, helpText);
      expect(root).toMatchSnapshot();
    });
  });

});

