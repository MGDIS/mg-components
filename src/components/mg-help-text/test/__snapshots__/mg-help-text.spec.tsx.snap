// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mg-help-text Should render text : Help text with <a href="https://www.exemple.com">a link</a>, a <strong>bold text</strong> and <em>an italic one</em>. renders 1`] = `
<mg-help-text>
  <mock:shadow-root>
    <p>
      <slot></slot>
    </p>
  </mock:shadow-root>
  Help text with &lt;a href="https://www.exemple.com"&gt;a link&lt;/a&gt;, a &lt;strong&gt;bold text&lt;/strong&gt; and &lt;em&gt;an italic one&lt;/em&gt;.
</mg-help-text>
`;

exports[`mg-help-text Should render text : Help text with <a href="https://www.exemple.com">a link</a>, a <strong>bold text</strong> and <em>an italic one</em>. renders 2`] = `
<mg-help-text>
  <mock:shadow-root>
    <p id="blu">
      <slot></slot>
    </p>
  </mock:shadow-root>
  Help text with &lt;a href="https://www.exemple.com"&gt;a link&lt;/a&gt;, a &lt;strong&gt;bold text&lt;/strong&gt; and &lt;em&gt;an italic one&lt;/em&gt;.
</mg-help-text>
`;

exports[`mg-help-text Should render text : Simple help text. renders 1`] = `
<mg-help-text>
  <mock:shadow-root>
    <p>
      <slot></slot>
    </p>
  </mock:shadow-root>
  Simple help text.
</mg-help-text>
`;

exports[`mg-help-text Should render text : Simple help text. renders 2`] = `
<mg-help-text>
  <mock:shadow-root>
    <p id="blu">
      <slot></slot>
    </p>
  </mock:shadow-root>
  Simple help text.
</mg-help-text>
`;
